docs/api/synchronization_library/barrier.md

```diff
@@ -44,34 +57,48 @@ auto ptr = barrier_native_handle(b);
 asm volatile (
     "mbarrier.arrive.b64 _, [%0];"
     :: "l"(ptr)
-    : "memory"); 
+    : "memory");
 // equivalent to: (void)b.arrive();
 ```
 
-## Omissions
-
-None.
-
 ## Restrictions
 
-An object of type `barrier` shall not be accessed concurrently by CPU and GPU threads unless:
-  1. it is in managed memory, with `concurrentManagedAccess==1`, or
-  2. it is in host memory, with `hostNativeAtomicSupported==1`.
+An object of type `barrier` shall not be accessed concurrently by CPU and GPU
+  threads unless:
+- it is in unified memory and the [`concurrentManagedAccess` property] is 1, or
+- it is in CPU memory and the [`hostNativeAtomicSupported` property] is 1.
 
-(Note, for objects of scopes other than `thread_scope_system` this is a data-race, and thefore also prohibited regardless of memory characteristics.)
+Note, for objects of scopes other than `thread_scope_system` this is a
+  data-race, and thefore also prohibited regardless of memory characteristics.
 
-Under Compute Capability 8 or above, when an object of type `barrier<thread_scope_block>` is placed in `__shared__` memory, the member function `arrive` performs a reduction of the arrival count among [coalesced](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#coalesced-group-cg) threads followed by the arrival operation in one thread. Programs shall ensure that this transformation would not introduce errors, for example relative to the requirements of `[thread.barrier.class]` paragraph 12 in N4860.
+Under CUDA Compute Capability 8 (Ampere) or above, when an object of type
+  `barrier<thread_scope_block>` is placed in `__shared__` memory, the member
```

Comment:
Ditto about "`__shared__`" here.

Reply:
I don't want to say "shared memory" because people who are NOT familiar with CUDA will read that as... well... the traditional meaning of shared memory: e.g. interprocess shared memory.